File: Models/Glasses.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Vision.Models
{
	public class Glasses
	{
        public Glasses()
        {
        }

        public Glasses(GlassesHistory g)
        {
            Group = g.Group;
            Number = g.Number;

            OD_Spherical = g.OD_Spherical;
            OD_Cylindrical = g.OD_Cylindrical;
            OD_Axis = g.OD_Axis;
            OD_Add = g.OD_Add;

            OS_Spherical = g.OS_Spherical;
            OS_Cylindrical = g.OS_Cylindrical;
            OS_Axis = g.OS_Axis;
            OS_Add = g.OS_Add;

            Sunglasses = g.Sunglasses;
            Size = g.Size;
            Gender = g.Gender;

            InsertDate = g.InsertDate;
        }

		[Key, Column(Order = 0)]
		public int Group { get; set; }
		[Key, Column(Order = 1)]
		public int Number { get; set; }

		public float OD_Spherical { get; set; }
		public float OD_Cylindrical { get; set; }
		public int OD_Axis { get; set; }
		public float OD_Add { get; set; }

		public float OS_Spherical { get; set; }
		public float OS_Cylindrical { get; set; }
		public int OS_Axis { get; set; }
		public float OS_Add { get; set; }

		public bool Sunglasses { get; set; }

		[StringLength(1)]
		public string Size { get; set; }

		[StringLength(1)]
		public string Gender { get; set; }

        public DateTime InsertDate { get; set; }
	}
}

File: src/main/java/com/andrewma/vision/models/Glasses.java

package com.andrewma.vision.models;

import com.andrewma.vision.database.core.DbDataType;
import com.andrewma.vision.database.core.annotations.Column;
import com.andrewma.vision.database.core.annotations.PrimaryKey;
import com.andrewma.vision.database.core.annotations.Table;

@Table
public class Glasses {
    @PrimaryKey
    public int GlassesId;

    @Column(dataType = DbDataType.INTEGER)
    public int Group;
    @Column(dataType = DbDataType.INTEGER)
    public int Number;

    @Column(dataType = DbDataType.INTEGER)
    public boolean OD_Blind;
    @Column(dataType = DbDataType.REAL)
    public float OD_Spherical;
    @Column(dataType = DbDataType.REAL)
    public float OD_Cylindrical;
    @Column(dataType = DbDataType.INTEGER)
    public int OD_Axis;
    @Column(dataType = DbDataType.REAL)
    public float OD_Add;

    @Column(dataType = DbDataType.INTEGER)
    public boolean OS_Blind;
    @Column(dataType = DbDataType.REAL)
    public float OS_Spherical;
    @Column(dataType = DbDataType.REAL)
    public float OS_Cylindrical;
    @Column(dataType = DbDataType.INTEGER)
    public int OS_Axis;
    @Column(dataType = DbDataType.REAL)
    public float OS_Add;

    @Column(dataType = DbDataType.TEXT, nullable = true)
    public Boolean Sunglasses;
    @Column(dataType = DbDataType.TEXT, nullable = true)
    public char Size;
    @Column(dataType = DbDataType.TEXT, nullable = true)
    public char Gender;

    /** Epoch Time (seconds since 1/1/1970 */
    @Column(dataType = DbDataType.INTEGER)
    public long AddedEpochTime;
    
    protected void copyFrom(Glasses glasses) {
        GlassesId = glasses.GlassesId;

        Group = glasses.Group;
        Number = glasses.Number;

        OD_Blind = glasses.OD_Blind;
        OD_Spherical = glasses.OD_Spherical;
        OD_Cylindrical = glasses.OD_Cylindrical;
        OD_Axis = glasses.OD_Axis;
        OD_Add = glasses.OD_Add;

        OS_Blind = glasses.OS_Blind;
        OS_Spherical = glasses.OS_Spherical;
        OS_Cylindrical = glasses.OS_Cylindrical;
        OS_Axis = glasses.OS_Axis;
        OS_Add = glasses.OS_Add;

        Sunglasses = glasses.Sunglasses;
        Size = glasses.Size;
        Gender = glasses.Gender;

        AddedEpochTime = glasses.AddedEpochTime;
    }
}
